### PR TITLE
Fix wildcard-in-the-middle in General-Electric.xml

### PR DESCRIPTION
**Pull request by RReverser**
See https://github.com/EFForg/https-everywhere/issues/4369

### CODE REVIEW

File: src/chrome/content/rules/General-Electric.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <ruleset name="General Electric (partial)">
 
-	<target host="files.*.geblogs.com"/>
+	<target host="*.geblogs.com"/>
```
Comment (J0WI) [inline src/chrome/content/rules/General-Electric.xml]: Could you also add some (we require at least 3) test urls and a hint, that the blogs itself do not support HTTPS.

Comment (RReverser) [reply src/chrome/content/rules/General-Electric.xml]: Doesn't rule itself give such a hint? Its regexp clearly covers only files.*.geblogs.com.

Comment (Bisaloo) [reply src/chrome/content/rules/General-Electric.xml]: This is the last PR needed before we can forbid wildcards in the middle. @RReverser, are you interested in addressing @J0WI's comments (ie add `test url`s and a comment *why* other subdomains do not support HTTPS)?

If you don't have time / don't want to work on this, I can take over.

**Comment by Bisaloo** [comment]
@J0WI, I would really like to move forward and finally block wildcards in the middle in our tests. Do you think it's okay if I edit this branch? Otherwise, I can also open a brand new PR for this.

**Comment by J0WI** [comment]
@Bisaloo since @RReverser is not responding I would say it's okay to take it.

File: src/chrome/content/rules/GEBlogs.com.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<!--
+	Invalid certificate:
+		*.files.geblogs.com
+		files.*.geblogs.com
+
+	Other subdomains are mismatched or refused.
```
Comment (J0WI) [inline src/chrome/content/rules/GEBlogs.com.xml]: @Bisaloo could you add some more documentation about the behavior of this host?
E.g. why a wildcard is required, but only specific patterns are rewritten.

File: src/chrome/content/rules/GEBlogs.com.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<!--
+	Invalid certificate:
+		*.files.geblogs.com
+		files.*.geblogs.com
```
Comment (J0WI) [inline src/chrome/content/rules/GEBlogs.com.xml]: `*.geblogs.com` are also mismatching, but see comment below.

File: src/chrome/content/rules/GEBlogs.com.xml (hunk 2 above)
Comment (Bisaloo) [reply src/chrome/content/rules/GEBlogs.com.xml]: How would you describe this? I don't know how to explain it better than what I did.

**Comment by J0WI** [comment]
@Bisaloo would you like to finish this?

**Comment by J0WI** [comment]
There are two wildcards in the `Invalid certificate` which says more or less the same as `Other subdomains are mismatched or refused.`. But there is also wildcard target, which makes it even more unclear. `geblogs.com` itself is not documented at all.

I would add a short description, that the service provides 2nd level subdomains for its customers and an additional `files` 3rd level subdomain for static content. For their own static content they use a subdomain with `files` in the 2nd and the category in the 3rd level.

Maybe it also helps to ad their "main" site?
```
ge.com
www.ge.com
benefits.ge.com
www.benefits.ge.com
ssologin.corporate.ge.com
ssousflogin.corporate.ge.com
digitalsupport.ge.com
go.digital.ge.com
eddie.ge.com
integrity.ge.com
libraries.ge.com
myapps.ge.com
mypassword.ge.com
mytech.ge.com
onehr.ge.com
www.onehr.ge.com
geaeck.registrar.ge.com
ssocentralck.registrar.ge.com
sc.ge.com
supportcentral.ge.com
travel.ge.com
```

**Comment by J0WI** [comment]
@Bisaloo this seems to be the last bit in https://github.com/EFForg/https-everywhere/issues/4369#issuecomment-326727385. Would you like to finish this?

**Comment by pipboy96** [comment]
:tada:

**Pull request merged**